src/PubSubClient.cpp

```diff
@@ -384,7 +383,7 @@ boolean PubSubClient::publish(const char* topic, const uint8_t* payload, unsigne
             return false;
         }
         // Leave room in the buffer for header and variable length field
-        uint16_t length = MQTT_MAX_HEADER_SIZE;
+        uint16_t length;
         length = writeString(topic,buffer,length);
```

Comment:
The value for `length` is used as a parameter here, so if you don't initialize it, it will stop working very soon.

Reply:
Done. Didn't notice that firstly it passed as argument. Now it should be more clear.